layouts/_default/list.html

```diff
@@ -159,6 +155,10 @@ <h2 class="text-2xl font-extrabold">
 {{ define "extra-main" }}
   {{ $p := partial "helper/paginator" . }}
   {{ $pages := $p.Pages }}
+  {{ $pinnedPages := where $pages "Params.pinned" true }}
+  {{ $featuredPages := union $pinnedPages $pages | first 5 }}
+  {{ $river := complement $featuredPages $pages }}
```

Comment:
Using `complement` here fixes the case of items from further down the river jumping to the top after pinning.